packages/indexer-agent/src/agent.ts

```diff
@@ -441,14 +447,26 @@ class Agent {
           )
 
           // Reconcile allocations
-          await this.reconcileAllocations(
-            activeAllocations,
-            targetAllocations,
-            indexingRules,
-            currentEpoch.toNumber(),
-            currentEpochStartBlock,
-            maxAllocationEpochs,
-          )
+          if (this.allocationManagementMode == AllocationManagementMode.AUTO) {
+            await this.reconcileAllocations(
+              activeAllocations,
+              targetAllocations,
+              indexingRules,
+              currentEpoch.toNumber(),
+              currentEpochStartBlock,
+              maxAllocationEpochs,
+            )
+          } else if (
+            this.allocationManagementMode == AllocationManagementMode.MANUAL
```

Comment:
It looks like `this.reconcileAllocations(...)` contains the code path for collecting receipts and exchanging them for a voucher. We need to expose this capability to the operator some how, or handle receipt collection in a separate loop. https://github.com/graphprotocol/indexer/blob/3813def52ab984e5c4d142a2607636eff2a737d4/packages/indexer-agent/src/agent.ts#L1039

Reply:
ok, so right now `collectReceipts()` is coupled with `closingAllocation()` in the agent reconciliation loop. 

I could certainly continue that pattern in the manual allocation management workflow by calling collect receipts when `indexer allocations close..` or `indexer allocations reallocate..` are used. However, this may be a good time to decouple closing from collecting and move `collectReceipts()` into a separate reconciliation loop function that has Eventual<recentlyClosedAllocations> as an input. 

In the case of allocationManagmentMode = manual, which behavior would you personally prefer? 
- A separate loop that uses the current threshold configs to automatically collect receipts for closed allocations. 
- Coupling collectReceipts with `allocations close` and `allocations reallocate` so the manual allocation actions lead to automatic managing of receipts. 
- Separate out collecting of receipts to a separate indexer-cli command, so indexers have to call it explicitly to collect receipts for an allocation.  If we did this I'd also include a command to get a summary of receipts per allocation to be a part of this workflow. Another option with this more manual workflow is to use a separate setting to control receipt collection; then we'd have both the `allocation-management` config and a separate config like `receipt-management` which can also be set to `auto` or `manual`.

Reply:
I think for this PR I'm going to translate the current close allocation behavior into the manual action, so I'll add the `collectReceipts()` execution at the end of the manual close allocation process.  
We can continue this conversation about how to move forward with the design and I'll implement the output of that discussion in a subsequent PR, likely with the action queue work.